File: api/java/indigo/src/main/java/com/epam/indigo/Hybridization.java
package com.epam.indigo;

public enum Hybridization {
    S(1, "S"),
    SP(2, "SP"),
    SP2(3, "SP2"),
    SP3(4, "SP3"),
    SP3D(5, "SP3D"),
    SP3D2(6, "SP3D2"),
    SP3D3(7, "SP3D3"),
    SP3D4(8, "SP3D4"),
    SP2D(9, "SP2D");

    private int numVal;
    private String strVal;

    Hybridization(int numVal, String strVal) {
        this.numVal = numVal;
        this.strVal = strVal;
    }

    public int getNumVal() {
        return numVal;
    }

    public String getStrVal() {
        return strVal;
    }


    public static Hybridization fromNum(int value) {
        switch (value) {
            case 1:
                return Hybridization.S;
            case 2:
                return Hybridization.SP;
            case 3:
                return Hybridization.SP2;
            case 4:
                return Hybridization.SP3;
            case 5:
                return Hybridization.SP3D;
            case 6:
                return Hybridization.SP3D2;
            case 7:
                return Hybridization.SP3D3;
            case 8:
                return Hybridization.SP3D4;
            case 9:
                return Hybridization.SP2D;
            default:
                throw new IndigoException(new Object(), "Unknown hybridization");
        }
    }
}

File: api/python/indigo/indigo/hybridization.py
# Copyright (C) from 2009 to Present EPAM Systems.
#
# This file is part of Indigo toolkit.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from enum import Enum


class Hybridization(Enum):
    S = 1
    SP = 2
    SP2 = 3
    SP3 = 4
    SP3D = 5
    SP3D2 = 6
    SP3D3 = 7
    SP3D4 = 8
    SP2D = 9
